pyro/poutine/__init__.py

```diff
@@ -23,10 +23,11 @@
 ############################################
 
 
-def trace(fn=None, graph_type=None):
+def trace(fn=None, graph_type=None, param_only=None):
```

Comment:
This arg is fine for now but would be awkward if we ever generalize. Do you think it would be reasonable to replace it with `sample_sites=True, param_sites=True`?

Also, do I understand correctly that even with `param_only=True`, the `_INPUT` and `_RETURN` sites will be recorded?

Also, why not simply `block` or filter to `{name: site for site in trace.nodes.items() if site['type'] == 'param'}`?

Reply:
> This arg is fine for now but would be awkward if we ever generalize. Do you think it would be reasonable to replace it with sample_sites=True, param_sites=True?

I don't see a use case for `sample_sites=True, param_sites=False`, so I don't know why we'd need to support that.

> Also, why not simply block 

This version has no effect on any enclosing code

> or filter

This way it's fine to wrap around something where a sample site would otherwise appear multiple times, e.g. a multi-particle ELBO where the model and guide are invoked repeatedly.